Sources/ArgumentParser/Parsing/ArgumentSet.swift

```diff
@@ -96,31 +96,20 @@ extension ArgumentSet {
   /// Creates an argument set for a single Boolean flag.
   static func flag(key: InputKey, name: NameSpecification, help: ArgumentHelp?) -> ArgumentSet {
     let help = ArgumentDefinition.Help(options: .isOptional, help: help, key: key)
-    let arg = ArgumentDefinition(kind: .name(key: key, specification: name), help: help, update: .nullary({ (origin, name, values) in
+    let arg = ArgumentDefinition(kind: .name(key: key, specification: name), environmentNames: name.makeEnvironmentNames(key), help: help, update: .nullary({ (origin, name, values) in
       values.set(true, forKey: key, inputOrigin: origin)
     }), initial: { origin, values in
       values.set(false, forKey: key, inputOrigin: origin)
     })
     return ArgumentSet(arg)
   }
 
-  static func updateFlag<Value: Equatable>(key: InputKey, value: Value, origin: InputOrigin, values: inout ParsedValues, hasUpdated: Bool, exclusivity: FlagExclusivity) throws -> Bool {
-    switch (hasUpdated, exclusivity) {
-    case (true, .exclusive):
-      // This value has already been set.
-      if let previous = values.element(forKey: key) {
-        if (previous.value as? Value) == value {
-          // setting the value again will consume the argument
-          values.set(value, forKey: key, inputOrigin: origin)
-        }
-        else {
-          throw ParserError.duplicateExclusiveValues(previous: previous.inputOrigin, duplicate: origin, originalInput: values.originalInput)
-        }
-      }
-    case (true, .chooseFirst):
-      values.update(forKey: key, inputOrigin: origin, initial: value, closure: { _ in })
-    case (false, _), (_, .chooseLast):
+  static func updateFlag<Value: Equatable>(key: InputKey, value: Value, origin: InputOrigin, values: inout ParsedValues, didUpdate: Bool, exclusivity: FlagExclusivity) throws -> Bool {
```

Comment:
I wonder whether it would be better to decouple `updateFlag` from the `exclusivity.shouldAllowUpdate` call, and just have it take the `ExclusivityResult` enum as an argument.